src/App.js

```diff
@@ -0,0 +1,162 @@
+    super(props);
+
+    this.state = {
+      todos: [
+        {
+          id: 1,
+          name: "Buy a Milk",
+          checked: true
+        },
+        {
+          id: 2,
+          name: "Daid will be read",
+          checked: true
+        },
+        {
+          id: 3,
+          name: "a",
+          checked: false
+        }
+      ],
+
+      inputAddValue: "",
+
+      message: ""
+    };
+  }
+
+  duplicateCheck = () => {
+    /* duplicateCheckValue oluşturuyorum + findIndex kullanıyorum öğrendiğim
+    kadarıyla bana string eşit değilse -1'den fazla bir numara veriyor bende bunu if elseden geçirerek
+    sonuca varıyorum.
+    */
+
+    const duplicateCheckValue = this.state.todos.findIndex(
+      todo => todo.name.toLowerCase() === this.state.inputAddValue.toLowerCase()
+    );
+
+    if (duplicateCheckValue > -1) {
+      this.setState({ message: "Mert Abi gets ANGRY!" });
+    } else {
+      //...this.state.todos dememin sebebi ise bunlardan sonrasına ekle anlamında object olduğu için {} kullandım
+      this.setState({
+        todos: [
+          ...this.state.todos,
+          { name: this.state.inputAddValue, id: Date.now(), checked: false }
+        ],
+        inputAddValue: "",
+        message: ""
+      });
+    }
+  };
+
+  inputValueFunc = event => {
+    this.setState({
+      inputAddValue: event.target.value
+    });
+  };
+
+  addBtn = () => {
+    //Eğer input'um boş geliyorsa hata mesajını setstate ediyorum yok sıkıntı yoksa duplicateCheck fonksiyonumu çağırıyorum
+    if (this.state.inputAddValue === "") {
+      this.setState({ message: "Fill in the Blank" });
+    } else {
+      this.duplicateCheck();
+    }
+  };
+
+  handleDontClick = name => {
+    //Filter kullanmamın sebebi, todo.name !== name koşulu sağlamadığında elemesi...
+    const checkDont = this.state.todos.map(todo => {
+      return todo.name === name ? { ...todo, checked: false } : todo;
+    });
+
+    this.setState({
+      todos: checkDont
+    });
+  };
+
+  handleClick = name => {
+    //*************todo.name === name  ? 'dden sonrasını mantıken anlamadım!
+    const checkClick = this.state.todos.map(todo => {
+      return todo.name === name ? { ...todo, checked: true } : todo;
+    });
+
+    this.setState({
+      todos: checkClick
+    });
+  };
+
+  handleDeleteTodo = name => {
+    const checkDelete = this.state.todos.filter(todo => todo.name !== name);
+
+    this.setState({
+      todos: checkDelete
+    });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>Todos App</h1>
+
+        <div className="AddTodo">
+          <Input
+            placeholder="Add to New Todo"
+            value={this.state.inputAddValue}
+            onChange={this.inputValueFunc}
+          />
+
+          <Button onclick={this.addBtn}>Add</Button>
+        </div>
+
+        <div className="error">{this.state.message}</div>
+
+        <div className="List">
+          <h1>dontCompleted</h1>
+
```

Comment:
Burada yeni bir fonksiyon olustur.
```suggestion
handleCompleteTodo=(e)=> {
this.handleClick(e.target.value)
}
```